vaadin-combo-box-flow-demo/src/main/java/com/vaadin/flow/component/combobox/demo/ComboBoxView.java

```diff
@@ -41,386 +51,314 @@
-            } else {
-                message.setText(
-                        "Selected artist: " + event.getValue().getArtist()
-                                + "\nThe old selection was: "
-                                + event.getOldValue().getArtist());
-            }
-        });
+        filteringComboBox.setItems(filter, elementsList);
+        filteringComboBox.setItemLabelGenerator(Elements::getName);
+        filteringComboBox.setClearButtonVisible(true);
         // end-source-example
-        //@formatter:on
+        addCard("Filtering", "Custom filtering", div, filteringComboBox);
 
-        comboBox.getStyle().set(ElementConstants.STYLE_WIDTH, WIDTH_STRING);
-        comboBox.setId("component-selection-box");
-        addCard("Using components",
-                "Rendering items using ComponentTemplateRenderer", comboBox,
-                message);
     }
 
-    private void createComboBoxWithInMemoryLazyLoading() {
+    private void customOptionsDemo() {
         // begin-source-example
-        // source-example-heading: Lazy loading between client and server
-        ComboBox<String> comboBox = new ComboBox<>();
+        // source-example-heading: Customizing drop down items with
+        // ComponentRenderer
+        ComboBox<Information> comboBox = new ComboBox<>();
+        comboBox.setLabel("User");
+        comboBox.setItems(
+                new Information("Gabriella",
+                        "https://randomuser.me/api/portraits/women/43.jpg"),
+                new Information("Rudi",
+                        "https://randomuser.me/api/portraits/men/77.jpg"),
+                new Information("Hamsa",
+                        "https://randomuser.me/api/portraits/men/35.jpg"),
+                new Information("Jacob",
+                        "https://randomuser.me/api/portraits/men/76.jpg"));
+
+        comboBox.setRenderer(new ComponentRenderer<>(information -> {
+            Div text = new Div();
+            text.setText(information.getText());
+
+            Image image = new Image();
+            image.setWidth("21px");
+            image.setHeight("21px");
+            image.setSrc(information.getImage());
+
+            FlexLayout wrapper = new FlexLayout();
+            text.getStyle().set("margin-left", "0.5em");
+            wrapper.add(image, text);
+            return wrapper;
+        }));
 
-        /*
-         * Using a large data set makes the browser request items lazily as the
-         * user scrolls down the overlay. This will also trigger server-side
-         * filtering.
-         */
-        List<String> names = getNames(500);
-        comboBox.setItems(names);
+        comboBox.setItemLabelGenerator(Information::getText);
         // end-source-example
 
-        comboBox.setId("lazy-loading-box");
-        addCard("Lazy Loading", "Lazy loading between client and server",
-                comboBox);
+        addCard("Presentation",
```

Comment:
For some reason the source code of this demo isn't visible on the demo page

Reply:
This seems to be because `// source-example-heading:` above is wrapped to two lines but only one line is parsed and it needs to match the card heading given for `addCard()`.

Reply:
This is now fixed.